internal/resolution/variablesources/installed_package.go

```diff
@@ -26,18 +29,20 @@ func (r *InstalledPackageVariableSource) GetVariables(ctx context.Context) ([]de
 		return nil, err
 	}
 
+	vr, err := mmsemver.NewConstraint(r.bundleVersion)
+	if err != nil {
+		return nil, err
+	}
+
 	// find corresponding bundle for the installed content
-	resultSet := catalogfilter.Filter(allBundles, catalogfilter.WithBundleImage(r.bundleImage))
+	resultSet := catalogfilter.Filter(allBundles, catalogfilter.And(
+		catalogfilter.WithPackageName(r.pkgName),
+		catalogfilter.WithName(r.bundleName),
+		catalogfilter.InMastermindsSemverRange(vr),
+	))
 	if len(resultSet) == 0 {
 		return nil, r.notFoundError()
 	}
-
-	// TODO: fast follow - we should check whether we are already supporting the channel attribute in the operator spec.
-	//       if so, we should take the value from spec of the operator CR in the owner ref of the bundle deployment.
-	//       If that channel is set, we need to update the filter above to filter by channel as well.
-	sort.SliceStable(resultSet, func(i, j int) bool {
-		return catalogsort.ByVersion(resultSet[i], resultSet[j])
-	})
```

Comment:
I think we no longer need sorting by version since we filter bundles by bundle version amongst other things. I assume previously we were doing sorting beucase multiple bundles could in theory use the same image (.e.g some sort of fake version).

Reply:
Got it. It looks like we check the result set for 0 bundles, but not more than 1. Can we add another check that errors if there is more than 1 bundle?